daemon/containerd/image.go

```diff
@@ -157,17 +157,32 @@ func (i *ImageService) GetImageManifest(ctx context.Context, refOrID string, opt
 
 	cs := i.client.ContentStore()
 
-	desc, err := i.resolveDescriptor(ctx, refOrID)
+	img, err := i.resolveImage(ctx, refOrID)
 	if err != nil {
 		return nil, err
 	}
 
+	desc := img.Target
 	if containerdimages.IsManifestType(desc.MediaType) {
+		plat := desc.Platform
+		if plat == nil {
+			config, err := img.Config(ctx, cs, platform)
+			if err != nil {
+				return nil, err
+			}
+			var configPlatform ocispec.Platform
+			if err := readConfig(ctx, cs, config, &configPlatform); err != nil {
+				return nil, err
```

Comment:
Should we have `cerrdefs.IsNotFound` here more and provide a better errors?
Currently that will result in `content digest X not found` error when requesting platform not present locally:

```bash
$ docker run --pull never --platform linux/s390x --rm -it busybox
docker: Error response from daemon: content digest sha256:7d602b12b1d9c1bdbf4c9255c0ba276ac0d9e0cd781a7c13461e4875cfcae509: not found.
See 'docker run --help'.
```